src/Core/Framework/Api/Controller/SalesChannelProxyController.php

```diff
@@ -412,18 +412,24 @@ private function updateCustomerToContext(string $customerId, SalesChannelContext
         $this->eventDispatcher->dispatch($event);
     }
 
-    private function persistPermissions(Request $request): void
+    private function persistPermissions(Request $request, SalesChannelContext $salesChannelContext): void
     {
-        $contextToken = $this->getContextToken($request);
+        $contextToken = $salesChannelContext->getToken();
 
-        $salesChannelId = (string) $request->request->get('salesChannelId');
```

Comment:
This is an BC. Why did you changed this?

Reply:
Because the SalesChannelContext is already present, so we can just pass it as a parameter.
And `persistPermissions` is private so it should not be breaking.

Reply:
Whoops you are right :)